### PR TITLE
tt: add 'TT_CLI_CFG' environmental variable

### LINKED ISSUE #813: Please respect TT_CLI_CFG environment variable as a path to tt.yaml
### Subject

I use the single `tt` binary in various `$PWD` directories, and sometimes the autodiscovery of `tt.yaml` works as expected, but other times it does not. Having to specify `--cfg` each time is not always convenient.

Could you please add a feature to retrieve the value from the `TT_CLI_CFG` environment variable as a path to `tt.yaml` and use it before the autodiscovery mechanism of `tt.yaml`?

The path of `tt.yaml` specified with the `--cfg` option must take priority over the `TT_CLI_CFG` environment variable. Therefore, the precedence should be as follows:
1. The `--cfg` option, if specified (must throw an error when the file is not readable).
2. The `TT_CLI_CFG` environment variable (works the same as `--cfg`).
3. The existing autodiscovery mechanism of `tt.yaml`.

`tt cfg dump` should indicate that the configuration was taken from the `TT_CLI_CFG` environment variable and print the resolved absolute path to `tt.yaml`.

### PR DESCRIPTION
**Pull request by better0fdead**
Added 'TT_CLI_CFG' environmental variable to specify the path to configuration file.

Closes #813

### CODE REVIEW

File: test/integration/cli/test_launch.py [hunk 1]
```diff
@@ -495,3 +495,47 @@ def test_std_err_stream_local_launch_non_existent_dir(tt_cmd, tmpdir):
     assert tt_process.returncode == 1
     assert "failed to change working directory" not in tt_process.stdout.readline()
     assert "failed to change working directory" in tt_process.stderr.readline()
+
+
+def test_launch_with_env_cfg(tt_cmd, tmpdir):
+    cmd = [tt_cmd, "cfg", "dump"]
+    # Create cfg in separate directory and set env.
+    tmpdir_with_env_config = tempfile.mkdtemp()
```
Comment (psergee) [inline test/integration/cli/test_launch.py:503]: Where is this dir removed?

File: test/integration/cli/test_launch.py [hunk 2]
```diff
@@ -495,3 +495,47 @@ def test_std_err_stream_local_launch_non_existent_dir(tt_cmd, tmpdir):
...
+    create_tt_config(tmpdir_with_env_config, tmpdir)
+    test_env = os.environ.copy()
+    test_env['TT_CLI_CFG'] = tmpdir_with_env_config + "/tt.yaml"
+
+    rc, output = run_command_and_get_output(cmd, cwd=os.getcwd(), env=test_env)
+    assert rc == 0
+    assert tmpdir_with_env_config + "/tt.yaml" in output
```
Comment (psergee) [inline test/integration/cli/test_launch.py:510]: We can write anything to the config and check the `dump` output.

**Review by psergee — COMMENTED** [review]
Thank you for the patch!

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,6 +5,13 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+## Unreleased
+
+### Added
+
+- `TT_CLI_CFG`: environmental variable to specify the path to the configuration file.
```
[the remark this answers is not in the document]
Comment (better0fdead) [reply CHANGELOG.md]: changed

File: README.md [hunk 4]
```diff
@@ -267,8 +267,13 @@ Taratool CLI can be launched in several modes:
 
 ### Configuration file
 
-By default, configuration file is named `tt.yaml`. With the `--cfg` flag
-you can specify the path to configuration file. Example of configuration
+By default, configuration file is named `tt.yaml`.
+
+You can specify the path to configuration file either by setting the `--cfg` flag or
+by setting `TT_CLI_CFG` environmental variable. `TT_CLI_CFG` environmental variable has lower
+priority than `--cfg` flag.
```
[the remark this answers is not in the document]
Comment (better0fdead) [reply README.md]: changed

File: cli/cmd/root.go [hunk 5]
```diff
@@ -259,6 +269,7 @@ func InitRoot() {
 		}
 	} else {
 		cmdCtx.Cli.ConfigDir = filepath.Dir(cmdCtx.Cli.ConfigPath)
+		log.Debugf("location of configuration file: '%s'", cmdCtx.Cli.ConfigPath)
```
[the remark this answers is not in the document]
Comment (better0fdead) [reply cli/cmd/root.go]: changed

File: test/integration/cli/test_launch.py (hunk 2 above)
Comment (better0fdead) [reply test/integration/cli/test_launch.py]: changed check

**Pull request merged**